core/src/main/java/com/orientechnologies/orient/core/storage/impl/local/OAbstractPaginatedStorage.java

```diff
@@ -1391,7 +1391,79 @@ public int getClusterIdByName(final String clusterName) {
     }
   }
 
+  public void preallocateRids(final OTransaction clientTx) {
+
+    try {
+      checkOpenness();
+      checkLowDiskSpaceRequestsAndBackgroundDataFlushExceptions();
+
+      @SuppressWarnings("unchecked")
+      final Iterable<ORecordOperation> entries = (Iterable<ORecordOperation>) clientTx.getAllRecordEntries();
+      final TreeMap<Integer, OCluster> clustersToLock = new TreeMap<>();
+
+      final Set<ORecordOperation> newRecords = new TreeSet<>(COMMIT_RECORD_OPERATION_COMPARATOR);
+
+      for (ORecordOperation txEntry : entries) {
+
+        if (txEntry.type == ORecordOperation.CREATED) {
+          newRecords.add(txEntry);
+          int clusterId = txEntry.getRID().getClusterId();
+          clustersToLock.put(clusterId, getClusterById(clusterId));
+        }
+      }
+      stateLock.acquireReadLock();
+      try {
+
+        checkOpenness();
+
+        makeStorageDirty();
+        atomicOperationsManager.startAtomicOperation((String) null, true);
+        try {
+          lockClusters(clustersToLock);
+
+          Map<ORecordOperation, OPhysicalPosition> positions = new IdentityHashMap<ORecordOperation, OPhysicalPosition>();
+          for (ORecordOperation txEntry : newRecords) {
+            ORecord rec = txEntry.getRecord();
+
+            if (rec.isDirty()) {
+              ORecordId rid = (ORecordId) rec.getIdentity().copy();
+              ORecordId oldRID = rid.copy();
+              final OCluster cluster = getClusterById(rid.getClusterId());
+              OPhysicalPosition ppos = cluster.allocatePosition(ORecordInternal.getRecordType(rec));
+              rid.setClusterId(cluster.getId());
+              positions.put(txEntry, ppos);
+              rid.setClusterPosition(ppos.clusterPosition);
+              clientTx.updateIdentityAfterCommit(oldRID, rid);
+            }
+          }
+          atomicOperationsManager.endAtomicOperation(false, null);
+        } catch (Exception e) {
+          atomicOperationsManager.endAtomicOperation(true, e);
+        }
+
+      } catch (IOException | RuntimeException ioe) {
+        throw OException.wrapException(new OStorageException("Could not preallocate RIDs"), ioe);
+      } finally {
+        stateLock.releaseReadLock();
+      }
+
+    } catch (RuntimeException ee) {
+      throw logAndPrepareForRethrow(ee);
+    } catch (Error ee) {
+      throw logAndPrepareForRethrow(ee);
+    } catch (Throwable t) {
+      throw logAndPrepareForRethrow(t);
+    }
+  }
   public List<ORecordOperation> commit(final OTransaction clientTx, Runnable callback) {
+    return  commit(clientTx,callback,false);
+  }
+
+  public List<ORecordOperation> commitPreAllocated(final OTransaction clientTx, Runnable callback) {
+    return commit(clientTx,callback,true);
+  }
+
+  private List<ORecordOperation> commit(final OTransaction clientTx, Runnable callback,boolean allocated) {
```

Comment:
`callback` never used.

Reply:
Callback is there since long time ago because it was used for indexes, it's not needed anymore anyway it need a bit bigger refactor to remove it, so I will keep as it is in this PR and will be removed in a future commit.

Reply:
Sure, I'm aware about that callbacks on public interfacing methods, but still there is no point in having unused parameters in private methods.

Reply:
I yep, i can get rid of it :)

Reply:
Thanks ;)